quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/appgw.bicep

```diff
@@ -0,0 +1,203 @@
+        }
+      ]
+      frontendIPConfigurations: array({
+        properties: {
+          publicIPAddress: {
+            id: appgwpip.id
+          }
+        }
+        name: 'appGatewayFrontendIP'
+      })
+      frontendPorts: [
+        {
+          name: 'appGatewayFrontendPort'
+          properties: {
+            port: 443
+          }
+        }
+      ]
+      backendAddressPools: [
+        {
+          name: 'defaultaddresspool'
+          properties: {
+            backendAddresses: [
+              {
+                ipAddress: backendIpAddress
+              }
+            ]
+          }
+        }
+      ]
+      backendHttpSettingsCollection: [
+        {
+          name: 'defaulthttpsetting'
+          properties: {
+            port: 80
+            protocol: 'Http'
+            cookieBasedAffinity: 'Disabled'
+            requestTimeout: 30
+            pickHostNameFromBackendAddress: true
+          }
+        }
+      ]
+      httpListeners: [
+        {
+          name: 'hlisten'
+          properties: {
+            frontendIPConfiguration: {
+              id: '${appgwResourceId}/frontendIPConfigurations/appGatewayFrontendIP'
+            }
+            frontendPort: {
+              id: '${appgwResourceId}/frontendPorts/appGatewayFrontendPort'
+            }
+            protocol: 'Https'
+            sslCertificate: {
+              id: frontendAgwCertificateId
+            }
+          }
+        }
+      ]
+      requestRoutingRules: [
+        {
+          name: 'appGwRoutingRuleName'
+          properties: {
+            ruleType: 'Basic'
+            priority: 100
+            httpListener: {
+              id: '${appgwResourceId}/httpListeners/hlisten'
+            }
+            backendAddressPool: {
+              id: '${appgwResourceId}/backendAddressPools/defaultaddresspool'
+            }
+            backendHttpSettings: {
+              id: '${appgwResourceId}/backendHttpSettingsCollection/defaulthttpsetting'
+            }
+          }
+        }
+      ]
+  }
+  dependsOn: [
+    rbacPropagationDelay
+  ]
+}
+output agwId string = agw.id
+output agwName string = agw.name
+
+resource appgwpip 'Microsoft.Network/publicIPAddresses@2020-07-01' = {
+  name: 'pip-${agwName}'
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+  }
+}
+output agwPip string = appgwpip.properties.ipAddress
+
+var keyVaultSecretsUserRole = resourceId('Microsoft.Authorization/roleDefinitions', '4633458b-17de-408a-b874-0445c86b69e6')
+resource kvAppGwSecretsUserRole 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = {
+  scope: akv
+  name: guid(resourceId('Microsoft.Network/applicationGateways',agwName), agwId.id, akv.id, keyVaultSecretsUserRole)
+  properties: {
+    roleDefinitionId: keyVaultSecretsUserRole
+    principalType: 'ServicePrincipal'
+    principalId: agwId.properties.principalId
+  }
+}
+
+module rbacPropagationDelay 'wait.bicep' = {
```

Comment:
why is this needed? since the AGW assignment has already taken place?  What step is left to "fail"?